orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -0,0 +1,354 @@
+    def cleanupConfig = AbstractDeployStrategyStage.CleanupConfig.fromStage(stage)
+
+    Map baseContext = [
+      (cleanupConfig.location.singularType()): cleanupConfig.location.value,
+      cluster                                : cleanupConfig.cluster,
+      moniker                                : cleanupConfig.moniker,
+      credentials                            : cleanupConfig.account,
+      cloudProvider                          : cleanupConfig.cloudProvider,
+    ]
+
+    if (stage.context.useSourceCapacity) {
+      stage.context.useSourceCapacity = false
+    }
+
+    // we expect a capacity object if a fixed capacity has been requested or as a fallback value when we are copying
+    // the capacity from the current server group
+    def savedCapacity = stage.context.savedCapacity ?: stage.context.capacity?.clone()
+    stage.context.savedCapacity = savedCapacity
+
+    // FIXME: this clobbers the input capacity value (if any). Should find a better way to request a new asg of size 0
+    stage.context.capacity = [
+      min    : 0,
+      max    : 0,
+      desired: 0
+    ]
+
+    def deploySteps = stageData.getDeploySteps()
+    if (deploySteps.isEmpty() || deploySteps[-1] != 100) {
+      deploySteps.add(100)
+    }
+
+    // Get source ASG from prior determineSourceServerGroupTask
+    def source = null
+
+    try {
+      StageData.Source sourceServerGroup
+
+      Stage parentCreateServerGroupStage = stage.directAncestors().find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
+
+      if (parentCreateServerGroupStage.status == ExecutionStatus.NOT_STARTED) {
+        // No point in composing the flow if we are called to plan "beforeStages" since we don't have any STAGE_BEFOREs.
+        // Also, we rely on the the source server group task to have run already.
+        // In the near future we will move composeFlow into beforeStages and afterStages instead of the
+        // deprecated aroundStages
+        return []
+      }
+
+      StageData parentStageData = parentCreateServerGroupStage.mapTo(StageData)
+      sourceServerGroup = parentStageData.source
+
+      if (sourceServerGroup != null && sourceServerGroup.serverGroupName != null) {
+        source = new ResizeStrategy.Source(
+          region: sourceServerGroup.region,
+          serverGroupName: sourceServerGroup.serverGroupName,
+          credentials: stageData.credentials ?: stageData.account,
+          cloudProvider: stageData.cloudProvider
+        )
+      }
+    } catch (Exception e) {
+      // This probably means there was no parent CreateServerGroup stage - which should never happen
+      throw new IllegalStateException("Failed to determine source server group from parent stage while planning Monitored Deploy flow", e)
+    }
+
+    MonitoredDeployStageData mdsd = stage.mapTo(MonitoredDeployStageData.class)
+
+    // TODO(mvulfson): I don't love this
+    def evalContext = mdsd.getChildStageContext()
```

Comment:
What is `mdsd`? Do I need to have a conversation with my doctor about it?

Reply:
**M**onitored**D**eploy**S**tage**D**ata, obviously! yeah, I don't love it, it shall be refactored, renamed in subsequent pr